spanner/client.go

```diff
@@ -280,6 +281,17 @@ func (c *Client) Single() *ReadOnlyTransaction {
 		t.sh = sh
 		return nil
 	}
+	if c.sc != nil {
+		_, instance, database, err := parseDatabaseName(c.sc.database)
```

Comment:
I don't think that we should parse this information every time a read-only transaction (or any other transaction) is requested from the client. Instead, it should be possible to just store the `CommonTags` in the `Client` (or the `sessionClient`) and assign that to each transaction that is created. The information in the `CommonTags` won't change during the lifetime of a client.